File: 05-LagrangeInterpolation/LagrangeInterpolation.py

def larange(xArr, yArr, n, x):
	sum = 0
	
	for i in range(n):
		mult = 1.0
		for j in range(n):
			if (i != j):
				mult = mult * (x - xArr[j]) / (xArr[i] - xArr[j])
		sum = sum + mult * yArr[i]
	
	return sum

n = int(input("Enter the number of degrees:  "))

xArray = list(n)
yArray = list(n)
print("Enter the value of x and y as \n")

for i in range(n):

	# xArray[i],  yArray[i] = int(input().split())
	
	xArray.append(int(input(f"x{i} = ")))
	yArray.append(int(input(f"y{i} = ")))


x = float(input("Enter the value of which the interpolation is required:  "))

answer = larange(xArray, yArray, n, x)

print(f"The  required answer is {answer}")

File: 05-LagrangeInterpolation/LagrangeInterpolation.c
#include<stdio.h>
#define MAX 10

float lagrangeMethod(float *xArr,float *yArr, int n, float x){
	float sum = 0;
	float mult;

	for (int i = 0; i < n; i++)
	{ 
		mult=1.0;
		for(int j = 0; j< n; j++)
		{ 
			if(i!=j)
				mult = mult * (x - *(xArr+j)) / (*(xArr+i) - *(xArr+j));
		}
		sum = sum +mult * yArr[i];
	}
	return sum;
}

int main()
{
	int n;
	float x1, y1;

	float x[MAX], y[MAX], mult, sum;

	printf("Enter the number of degrees:   ");
	scanf("%d", &n);


	printf("Enter the values of x and y\n");
	printf("x | y\n");

	for (int i = 0; i < n; i++)
	{
		scanf("%f	%f", &x[i], &y[i]);
	}

	printf("Enter x where the interpolation is required: ");
	scanf("%f", &x1);

	y1 = lagrangeMethod(x, y, n, x1);

	printf("y= %.4f\n", y1);
}